stat/distuv/noncentralt.go

```diff
@@ -0,0 +1,317 @@
+// Copyright ©2025 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"errors"
+	"math"
+	"math/rand/v2"
+
+	"gonum.org/v1/gonum/mathext"
+)
+
+// NoncentralT is the noncentral t-distribution.
+//
+// See https://en.wikipedia.org/wiki/Noncentral_t-distribution for more details.
+type NoncentralT struct {
+	// Nu is the degrees of freedom.
+	Nu float64
+
+	// Ncp is the noncentral parameter.
+	Ncp float64
+
+	// Src is the random source used to generate samples.
+	Src rand.Source
+}
+
+// Rand samples from the noncentral t-distribution.
+func (dist NoncentralT) Rand() float64 {
+	n := dist.Ncp + rand.New(dist.Src).NormFloat64()
+	c2 := ChiSquared{K: dist.Nu, Src: dist.Src}.Rand()
+	return n / math.Sqrt(c2/dist.Nu)
+}
+
+// Mean returns the mean of the noncentral t-distribution.
+func (dist NoncentralT) Mean() float64 {
+	nu := dist.Nu
+	if nu <= 1 {
+		return math.NaN()
+	}
+	return dist.Ncp * math.Sqrt(nu/2) * gammaADivB((nu-1)/2, nu/2)
+}
+
+// Variance returns the variance of the noncentral t-distribution.
+func (dist NoncentralT) Variance() float64 {
+	nu := dist.Nu
+	if nu <= 2 {
+		return math.NaN()
+	}
+	mean := dist.Mean()
+	return nu*(1+dist.Ncp*dist.Ncp)/(nu-2) - mean*mean
+}
+
+// Prob returns the probability density function of the noncentral t-distribution.
+func (dist NoncentralT) Prob(x float64) float64 {
+	return math.Exp(dist.LogProb(x))
+}
+
+// LogProb returns the log-probability density function of the noncentral t-distribution.
+func (dist NoncentralT) LogProb(x float64) float64 {
+	var epsilon = math.Nextafter(1, 2) - 1
+	ax := math.Abs(x)
+	if ax > math.Sqrt(dist.Nu*epsilon) {
+		a := NoncentralT{Nu: dist.Nu + 2, Ncp: dist.Ncp}.CDF(x * math.Sqrt(1+2/dist.Nu))
+		b := dist.CDF(x)
+		return math.Log(dist.Nu) - math.Log(ax) + math.Log(math.Abs(a-b))
+	} else {
+		return lgamma((dist.Nu+1)/2) - lgamma(dist.Nu/2) - 0.5*(math.Log(math.Pi)+math.Log(dist.Nu)+dist.Ncp*dist.Ncp)
+	}
```

Comment:
```suggestion
	}
	return lgamma((dist.Nu+1)/2) - lgamma(dist.Nu/2) - 0.5*(math.Log(math.Pi)+math.Log(dist.Nu)+dist.Ncp*dist.Ncp)
```
We should also be able to provide a constant for `math.Log(math.Pi)`.